### PR TITLE
Support C# 8 target-type new expressions.

### PR DESCRIPTION
**Pull request by patrickt**
Not quite sure whether this is the best place to do this, but
modifying `object_creation_expression` introduced conflicts,
and this seems to work without them.

### CODE REVIEW

**Comment by damieng** [comment]
I tried this in the latest VS beta that has C# 9.0 support:

```csharp
static void b(int a, int b)
{
    var x = new(a, b);
}
```

The generated syntax tree was:

image

File: grammar.js [hunk 1]
```diff
@@ -1059,10 +1066,10 @@ module.exports = grammar({
 
     interpolation_format_clause: $ => seq(':', /[^}"]+/),
 
-    invocation_expression: $ => prec(PREC.INVOCATION, seq(
+    invocation_expression: $ => seq(
```
Comment (initram) [inline grammar.js:1069]: It seems that you have accidentally removed the precedence value I added here, that is why the test is failing.

**Comment by damieng** [comment]
I think we should have named this `implicit_object_creation_expression` rather than `target_type_new_object_creation_expression` so as to stay consistent with Roslyn (sorry, my comment with the screenshot wasn't very clear)

**Comment by patrickt** [comment]
I took another run at this in #97. I’ll see if we can get some better naming in the resulting tree.

**Pull request closed without merging**